src/unitxt/artifact.py

```diff
@@ -557,3 +557,16 @@ def get_artifacts_data_classification(artifact: str) -> Optional[List[str]]:
         return None
 
     return data_classification.get(artifact)
+
+
+def return_artifact_specific_kwargs(
+    artifact: Artifact, keep_empty: bool = True
+) -> Dict[str, Any]:
+    attributes = artifact.__annotations__.keys()
+    kwargs = {}
+    for attribute in attributes:
+        value = getattr(artifact, attribute)
+        if not keep_empty and value is None:
+            continue
+        kwargs[attribute] = value
+    return kwargs
```

Comment:
Given this is the functionality and the fact it uses Dataclass internals I think it shuold be a function of Dataclass.
For the benefit of future users it might be good if it was a functionality of `to_dict` something like `to_dict(without_parent=True)`. What do you think @pawelknes ?

Reply:
@elronbandel - do you mean that it would be a function of Artifact called to_dict()?

Reply:
Yup, that's good idea

Reply:
I've moved the method to `Dataclass` and also modified it so that it is now able to return parameters of chosen specified parent classes (maybe that will be useful in future)

And I've also made `InferenceEngineParams` classes to be Mixin now as you suggested